website/docs/r/supervisor_namespace.html.markdown

```diff
@@ -0,0 +1,152 @@
+---
+layout: "vcfa"
+page_title: "VMware Cloud Foundation Automation: vcfa_supervisor_namespace"
+sidebar_current: "docs-vcfa-resource-supervisor-namespace"
+description: |-
+ Provides a VMware Cloud Foundation Automation Supervisor Namespace. This can be used to create, modify, and delete Roles.
+---
+
+# vcfa\_supervisor\_namespace
+
+Provides a VMware Cloud Foundation Automation Supervisor Namespace. This can be used to create, modify, and delete Roles.
+
+## Example Usage
+
+```hcl
+resource "vcfa_supervisor_namespace" "supervisor_namespace" {
+  name_prefix  = "terraform-demo"
+  project_name = "default-project"
+  class_name   = "small"
+  description  = "Supervisor Namespace created by Terraform"
+  region_name  = "default-region"
+  vpc_name     = "default-vpc"
+
+  storage_classes_initial_class_config_overrides {
+    limit_mib = 10000
+    name      = "vSAN Default Storage Policy"
+  }
+
+  zones_initial_class_config_overrides {
+    cpu_limit_mhz          = 1000
+    cpu_reservation_mhz    = 0
+    memory_limit_mib       = 1000
+    memory_reservation_mib = 0
+    name                   = "default-zone"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name_prefix` (Required) Prefix for the Supervisor Namespace Name. It must match RFC 1123 Label name (lower-case alphabet,
+  numbers between 0 and 9 and hyphen `-`)
+* `project_name` - (Required) The name of the Project where the Supervisor Namespace belongs to
+* `class_name` - (Required) The name of the Supervisor Namespace Class
+* `description` - (Optional) Description
+* `storage_classes_initial_class_config_overrides` - (Required) A set of Supervisor Namespace Storage Classes Initial Class Config Overrides. At least one is required. See [Storage Classes Initial Class Config Overridess](#storage-classes-initial-class-config-overrides) section for details
+* `region_name` - (Required)Name of the Region
+* `vpc_name` - (Required) Name of the VPC
+* `zones_initial_class_config_overrides` - (Required) A set of Supervisor Namespace Zones Initial Class Config Overrides. At least one is required. See [Zones Initial Class Config Overrides](#zones-initial-class-config-overrides) section for details
+
+## Attribute Reference
+
+* `name` - The name of the Supervisor Namespace
+- `phase` - Phase of the Supervisor Namespace
+- `ready` - Whether the Supervisor Namespace is in a ready status or not
+- `storage_classes` - A set of Supervisor Namespace Storage Classes. See [Storage Classes](#storage-classes) section for details
+- `vm_classes` - A set of Supervisor Namespace VM Classes. See [VM Classes](#vm-classes) section for details
+- `zones` - A set of Supervisor Namespace Zones. See [Zones](#zones) section for details
+
+<a id="storage-classes"></a>
+## Storage Classes
+
+The `storage_classes` is a set of metadata entries that have the following structure:
```

Comment:
I see "metadata entries" in every block. Is it because it's `metadata` inside K8s YAML or is it a typo?

Reply:
No, it has nothing to do with the k8s metadata. This may be a leftover when I copied and pasted from other vcd resources that contain a `metadata_entries` property. Removing all `metadata` from this file (and the resource) is better.

Reply:
Done.